src/main/java/org/truffleruby/language/TruffleBootNodes.java

```diff
@@ -263,19 +264,25 @@ public DynamicObject originalArgv() {
 
     }
 
-    @CoreMethod(names = "original_load_path", onSingleton = true)
-    public abstract static class OriginalLoadPathNode extends CoreMethodNode {
+    @CoreMethod(names = "extra_load_paths", onSingleton = true)
+    public abstract static class ExtraLoadPathSNode extends CoreMethodNode {
 
         @Child private StringNodes.MakeStringNode makeStringNode = StringNodes.MakeStringNode.create();
 
         @TruffleBoundary
         @Specialization
-        public DynamicObject originalLoadPath() {
+        public DynamicObject extraLoadPaths() {
             final String[] paths = getContext().getOptions().LOAD_PATHS;
             final Object[] array = new Object[paths.length];
 
             for (int n = 0; n < array.length; n++) {
-                array[n] = makeStringNode.executeMake(paths[n], UTF8Encoding.INSTANCE, CodeRange.CR_UNKNOWN);
+                final String path;
+                try {
+                    path = new File(paths[n]).getCanonicalPath();
```

Comment:
We could use `File.expand_path` I guess, but I'm a bit afraid of the cost on startup.
Probably small if `-I` is not used, so that might be the better option.

Reply:
I don't know if there are real problems to resolve, but it does diverge from MRI. I think this is the issue I encountered when I last looked at changing this.

I suppose to the extent that we want path matches, it's far more common for someone to do something like `-I $PWD/linked_dir` than it is for them to do `-I $(readlink "$PWD/linked_dir")`.

Reply:
OK, I pushed the change to use File.expand_path and added a spec.
I actually tried getAbsolutePath/getCanonicalPath for what does it do with `~/...` and `~eregon/...` (only the first one is handled), but didn't think about symlinks, good catch!